### ISSUE TITLE
E3002 - NAT gateway secondary allocation throws error

### ISSUE THREAD

**Issue by JihedTounsiTR** [aws-cloudformation/cfn-lint#2825]
### CloudFormation Lint Version

cfn-lint 0.72.2

### What operating system are you using?

Mac, Ubuntu and AL2

### Describe the bug

Hello 👋, 

When trying to add multiple IPs to NAT gateway following the [official documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-natgateway.html#cfn-ec2-natgateway-secondaryallocationids).
```yaml
Type: AWS::EC2::NatGateway
Properties: 
  AllocationId: String
  ConnectivityType: String
  MaxDrainDurationSeconds: Integer
  PrivateIpAddress: String
  SecondaryAllocationIds: 
    - String
  SecondaryPrivateIpAddressCount: Integer
  SecondaryPrivateIpAddresses: 
    - String
  SubnetId: String
  Tags: 
    - Tag
 ```
The linter throws an `E3002 Invalid Property Resources/NATGatewayAZOne/Properties/SecondaryAllocationIds` 💥

I have successfully updated the template and now my NAT GWs have extra public IPs, I just had to turn off the linting step in my pipeline.

Many thanks 😃

 

### Expected behavior

This is a legitimate property of the NAT gateway and the linter should not throw the `E3002`. 

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Resources:
  NATGatewayEIPAZOne:
    Type: AWS::EC2::EIP
    Properties:
      Domain: vpc

  NATGatewayEIPAZOneSecondary:
    Type: AWS::EC2::EIP
    Properties:
      Domain: vpc

  ProxyNATGatewayAZOne:
    Type: AWS::EC2::NatGateway
    Properties:
      AllocationId: !GetAtt NATGatewayEIPAZOne.AllocationId
      SecondaryAllocationIds:
        - !GetAtt NATGatewayEIPAZOneSecondary.AllocationId
      ConnectivityType: public
      SubnetId: !Ref NATGatewaySubnetAZOne # Subnet declaration ommited
```

**Comment by kddejong** [comment]
This is resolved in cfn-lint>=v0.73.1.  You can do a pip update to get the newest version.  

If you need to stick on the version you are at run `cfn-lint --update-specs`
